yt_dlp/extractor/tubetugraz.py

```diff
@@ -0,0 +1,349 @@
+            "title": "#6 (23.11.2017)",
+            "episode": "#6 (23.11.2017)",
+            "series": "[INB03001UF] Einführung in die strukturierte Programmierung",
+            "creator": "Safran C",
+            "duration": 3295818,
+        }
+    }
+
+    _LOGIN_REQUIRED = False
+    _NETRC_MACHINE = None
+
+    _LOGIN_URL = "https://tube.tugraz.at/Shibboleth.sso/Login?target=/paella/ui/index.html"
+    _LOGIN_SUCCESS_URL = "https://tube.tugraz.at/paella/ui/index.html"
+
+    _API_SERIES = "https://tube.tugraz.at/series/series.json"
+    _API_EPISODE = "https://tube.tugraz.at/search/episode.json"
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            return
+
+        result = self._download_webpage_handle(
+            self._LOGIN_URL, None,
+            note='downloading login page',
+            errnote='unable to fetch login page',
+            fatal=False)
+        if result is False:
+            return
+        else:
+            _, login_page_handle = result
+
+        result = self._download_webpage_handle(
+            login_page_handle.url, None,
+            note='logging in',
+            errnote='unable to log in',
+            fatal=False,
+            data=urlencode_postdata({
+                b"lang": "de",
+                b"_eventId_proceed": "",
+                b"j_username": username,
+                b"j_password": password
+            }),
+            headers={
+                "referer": login_page_handle.url
+            })
+        if result is False:
+            return
+        else:
+            _, result_page_handle = result
+
+        if result_page_handle.url != self._LOGIN_SUCCESS_URL:
+            self.report_warning("unable to login: incorrect password")
+            return
+
+    def _real_initialize(self):
+        self._login()
+
+    def _real_extract(self, url):
+        match = re.match(self._VALID_URL, url)
+
+        id = match.group('id')
+        if match.group('series') is not None:
+            return self._extract_series(id)
+        elif match.group('episode') is not None:
+            return self._extract_episode(id)
+        else:
+            raise ExtractorError("no video found on page")
+
+    def _extract_series(self, id):
+        series_info_data = self._download_json(
+            self._API_SERIES, None,
+            note='downloading series metadata',
+            errnote='failed to download series metadata',
+            fatal=False,
+            query={
+                "seriesId": id,
+                "count": 1,
+                "sort": "TITLE"
+            })
+        series_info = try_get(
+            series_info_data,
+            lambda x: x["catalogs"][0]["http://purl.org/dc/terms/"]) or {}
+
+        if len(series_info) == 0:
+            self.report_warning(
+                "failed to download series metadata: "
+                + "authentication required or series does not exist", id)
```

Comment:
why warning? shouldn't this be an error?

Also, 
1. don't use `+`
2. `if not series_info`

Reply:
if the warning message is too long, is it okay if it overshoots the 80 character limit in order to not use a `+`?

Reply:
Yes